nvflare/fuel/hci/client/cli.py

```diff
@@ -370,6 +389,15 @@ def _do_default(self, line):
             self.write_string(self.api.shutdown_msg)
             return True
 
+    def preloop(self):
+        if readline and os.path.exists(self.cli_history_file):
+            readline.read_history_file(self.cli_history_file)
+
+    def postcmd(self, stop, line):
+        if readline:
+            readline.write_history_file(self.cli_history_file)
+        return stop
```

Comment:
Why are we using postcmd here?
Can we use postloop?

Reply:
If it crashes before exiting the loop, then postloop() won't be called and history won't be saved